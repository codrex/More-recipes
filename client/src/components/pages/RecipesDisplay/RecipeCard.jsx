import React from 'react';
import PropTypes from 'prop-types';
import { Image } from 'cloudinary-react';
import classnames from 'classnames';
import Icon from '../../common/Icon';
import Button from '../../common/Button';
import Animation from '../../common/Animation';
import { DEFAULT_RECIPE_PIX } from '../../../constants';


const RecipeCard = (props) => {
  const fav = props.isFav(props.recipeId);
  const {
    recipeName,
    category,
    recipeId,
    downVotes,
    upVotes,
    views,
    push,
    image
  } = props;
  return (
    <Animation>
      <div className="recipe-card">
        <div className="recipe-card-body">
          <Image
            cloudName="resycom"
            publicId={image.publicId || DEFAULT_RECIPE_PIX}
            width="400"
            height="250"
            crop="fill"
            className="recipe-card-img"
          />
          <div className="recipe-card-underlay">
            <p className="lead text-capitalize">{category}</p>
            <Button
              handleClick={() => {
                push(`/recipe/${recipeId}`);
              }}
              text="view details"
              className="recipe-card-underlay-btn"
            />
          </div>
        </div>
        <div className="recipe-card-footer">
          <h4 className="ellipsis">{recipeName}</h4>
          <div className="d-flex col-12 no-padding justify-content-between">
            <div className="d-flex col-8 no-padding">
              <Icon iconClass="fa fa-thumbs-up recipe-card-icon ">
                {upVotes}
              </Icon>
              <Icon iconClass="fa fa-thumbs-down recipe-card-icon ">
                {downVotes}
              </Icon>
              <Icon iconClass="fa fa-eye recipe-card-icon">{views}</Icon>
            </div>
            <Icon
              iconClass={classnames('fa recipe-card-icon fav-color fav',
                !fav && 'fa-heart-o', fav && 'fa-heart')}
              pointer
              handleClick={() => props.toggleFav(recipeId)}
            />
          </div>
        </div>
      </div>
    </Animation>
  );
};

RecipeCard.propTypes = {
  upVotes: PropTypes.number.isRequired,
  downVotes: PropTypes.number.isRequired,
  views: PropTypes.number.isRequired,
  recipeName: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  recipeId: PropTypes.number.isRequired,
  toggleFav: PropTypes.func.isRequired,
  isFav: PropTypes.func.isRequired,
  push: PropTypes.func.isRequired,
  image: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
};

RecipeCard.defaultProps = {
  image: {}
};

export default RecipeCard;
