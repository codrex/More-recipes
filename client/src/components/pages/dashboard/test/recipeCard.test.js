import React from 'react';
import RecipeCard from '../recipeCard';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';


describe('Recipes card component ', () => {
  test('render as expected', () => {
    const component = renderer.create(
      <BrowserRouter>
        < RecipeCard
          recipeName="rice and beans"
          category="dinner"
          view={120}
          upVotes={30}
          downVotes={0}
        />
      </BrowserRouter>
  );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
