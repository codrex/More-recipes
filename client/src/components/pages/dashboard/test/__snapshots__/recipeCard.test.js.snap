// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes card component  render as expected 1`] = `
<div
  className="recipe-card"
>
  <div
    className="recipe-card-body"
  >
    <img
      alt="rice and beans"
      className="recipe-card-img"
      src="https://i.pinimg.com/564x/b5/be/08/b5be084a849fa67c190cc85b1e9216c3.jpg"
    />
    <div
      className="recipe-card-underlay"
    >
      <p
        className="lead"
      >
        dinner
      </p>
      <h4
        className="recipe-card-title"
      >
        rice and beans
      </h4>
      <a
        href="recipe/undefined"
        onClick={[Function]}
      >
        <button
          className="btn text-capitalize recipe-card-underlay-btn"
          data-dismiss="modal"
          data-target={undefined}
          data-toggle={undefined}
          onClick={undefined}
          type="button"
        >
          view details
        </button>
      </a>
    </div>
  </div>
  <div
    className="recipe-card-footer"
  >
    <div
      className="icon-wrapper"
      onClick={undefined}
      onMouseEnter={undefined}
    >
      <i
        className="icon fa fa-thumbs-up recipe-card-icon "
        data-target={undefined}
        data-toggle={undefined}
      >
        <span>
          30
        </span>
      </i>
    </div>
    <div
      className="icon-wrapper"
      onClick={undefined}
      onMouseEnter={undefined}
    >
      <i
        className="icon fa fa-thumbs-down recipe-card-icon "
        data-target={undefined}
        data-toggle={undefined}
      >
        <span>
          0
        </span>
      </i>
    </div>
    <div
      className="icon-wrapper"
      onClick={undefined}
      onMouseEnter={undefined}
    >
      <i
        className="icon fa fa-eye recipe-card-icon"
        data-target={undefined}
        data-toggle={undefined}
      >
        <span />
      </i>
    </div>
    <div
      className="icon-wrapper pointer"
      onClick={undefined}
      onMouseEnter={undefined}
    >
      <i
        className="icon fa fa-heart-o recipe-card-icon fav-color"
        data-target={undefined}
        data-toggle={undefined}
      >
        <span />
      </i>
    </div>
  </div>
</div>
`;
